iis-wf/iis-wf-report/src/main/java/eu/dnetlib/iis/wf/report/pushgateway/converter/GaugesBuilder.java

```diff
@@ -0,0 +1,56 @@
+package eu.dnetlib.iis.wf.report.pushgateway.converter;
+
+import eu.dnetlib.iis.common.java.stream.ListUtils;
+import io.prometheus.client.Gauge;
+
+import java.util.List;
+
+public class GaugesBuilder {
+
+    private GaugesBuilder() {
+    }
+
+    @FunctionalInterface
+    public interface BuilderWithLabels {
+        Gauge build(String metricName,
+                    String help,
+                    List<String> labelNames,
+                    List<List<String>> labelValues,
+                    List<Double> values);
+    }
+
+    public static Gauge buildGaugeWithLabels(String metricName,
+                                             String help,
+                                             List<String> labelNames,
+                                             List<List<String>> labelValues,
+                                             List<Double> values) {
+        Gauge gauge = Gauge.build()
+                .name(metricName)
+                .help(help)
+                .labelNames(labelNames.toArray(new String[0]))
+                .create();
+        ListUtils.zip(labelValues, values).forEach(labelValuesAndValue ->
+                gauge.labels(labelValuesAndValue.getLeft().toArray(new String[0])).set(labelValuesAndValue.getRight())
```

Comment:
Same here, we could initialize `String` array with an expected size.

Reply:
https://github.com/openaire/iis/pull/1072#discussion_r426593666